orderBook.js

```diff
@@ -0,0 +1,66 @@
+function reconcileOrder (existingBook, incomingOrder) {
```

Comment:
Having everything in a single function gets very messy. Remember, functions should do one thing. Review the solution branch to see how to better break down the problem into pieces.